test/e2e/upgrades/ingress.go

```diff
@@ -84,13 +112,15 @@ func (t *IngressUpgradeTest) Test(f *framework.Framework, done <-chan struct{},
 		// Ingress. Restarting the ingress controller and deleting ingresses
 		// while it's down will leak cloud resources, because the ingress
 		// controller doesn't checkpoint to disk.
-		t.verify(f, done, true)
+		t.verify(f, done, true, false)
+	case IngressUpgrade:
+		t.verify(f, done, true, true)
 	default:
 		// Currently ingress gets disrupted across node upgrade, because endpoints
 		// get killed and we don't have any guarantees that 2 nodes don't overlap
 		// their upgrades (even on cloud platforms like GCE, because VM level
 		// rolling upgrades are not Kubernetes aware).
-		t.verify(f, done, false)
+		t.verify(f, done, false, false)
```

Comment:
I think we're okay with always triggering a sync regardless of being a master or ingress upgrade.

Reply:
We can enable for master upgrades later if you prefer.

Reply:
Removed triggerSync bool so its enabled for master and ingress upgrade.